cmds/contest/server/server.go

```diff
@@ -61,7 +62,74 @@ func initFlags(cmd string) {
 			"This is the maximum amount of time a job can stay paused safely.")
 }
 
-func ServerMain(cmd string, args []string, sigs <-chan os.Signal) error {
+// TargetManagerConfig contains the target manager configuration.
+type TargetManagerConfig struct {
+	Name    string
+	Factory target.TargetManagerFactory
+}
+
+// TestFetcherConfig contains the test fetcher configuration.
+type TestFetcherConfig struct {
+	Name    string
+	Factory test.TestFetcherFactory
+}
+
+// TestStepConfig contains the test step configuration.
+type TestStepConfig struct {
+	Name    string
+	Factory test.TestStepFactory
+	Events  []event.Name
+}
+
+// ReporterConfig contains the reporter configuration.
+type ReporterConfig struct {
+	Name    string
+	Factory job.ReporterFactory
+}
+
+// PluginConfig contains the configuration for all the plugins desired in a
+// server instance.
+type PluginConfig struct {
+	TargetManagers []TargetManagerConfig
+	TestFetchers   []TestFetcherConfig
+	TestSteps      []TestStepConfig
+	Reporters      []ReporterConfig
+}
+
+func registerPlugins(pluginRegistry *pluginregistry.PluginRegistry, pluginConfig *PluginConfig) error {
```

Comment:
Almost everything in this file already exists in `cmds/plugins/plugins.go` (which is the wrong place now that I think of it)

Reply:
I'll see what I can reuse, can't remember now why I recreated it

Reply:
done